sonar-ruby-plugin/src/main/java/org/sonarsource/ruby/externalreport/rubocop/RuboCopSensor.java

```diff
@@ -0,0 +1,120 @@
+import org.sonarsource.analyzer.commons.ExternalRuleLoader;
+import org.sonarsource.analyzer.commons.internal.json.simple.parser.ParseException;
+
+public class RuboCopSensor implements Sensor {
+
+  private static final Logger LOG = Loggers.get(RuboCopSensor.class);
+
+  static final String LINTER_KEY = "rubocop";
+
+  static final String LINTER_NAME = "RuboCop";
+
+  public static final String REPORT_PROPERTY_KEY = "sonar.ruby.rubocop.reportPaths";
+
+  @Override
+  public void describe(SensorDescriptor descriptor) {
+    descriptor
+      .onlyWhenConfiguration(conf -> conf.hasKey(REPORT_PROPERTY_KEY))
+      .name("Import of RuboCop issues");
+  }
+
+  @Override
+  public void execute(SensorContext context) {
+    List<File> reportFiles = ExternalReportProvider.getReportFiles(context, REPORT_PROPERTY_KEY);
+    reportFiles.forEach(report -> importReport(report, context));
+  }
+
+  private static void importReport(File reportPath, SensorContext context) {
+    try (InputStream in = new FileInputStream(reportPath)) {
+      LOG.info("Importing {}", reportPath);
+      RuboCopJsonReportReader.read(in, issue -> saveIssue(context, issue));
+    } catch (IOException | RuntimeException | ParseException e) {
+      LOG.error("No issues information will be saved as the report file '{}' can't be read. " + e.getMessage(), reportPath, e);
+    }
+  }
+
+  private static void saveIssue(SensorContext context, RuboCopJsonReportReader.Issue issue) {
+    if (isEmpty(issue.ruleKey) || isEmpty(issue.filePath) || isEmpty(issue.message)) {
+      LOG.debug("Missing information or unsupported file type for ruleKey:'{}', filePath:'{}', message:'{}'",
+        issue.ruleKey, issue.filePath, issue.message);
+      return;
+    }
+    FilePredicates predicates = context.fileSystem().predicates();
+    InputFile inputFile = context.fileSystem().inputFile(predicates.hasPath(issue.filePath));
+    if (inputFile == null) {
+      LOG.warn("No input file found for {}. No RuboCop issues will be imported on this file.", issue.filePath);
+      return;
+    }
+    RuleKey qualifiedRuleKey = RuleKey.of(LINTER_KEY, issue.ruleKey);
+    NewExternalIssue newExternalIssue = context.newExternalIssue();
+    String ruleKey = qualifiedRuleKey.rule();
+
+    ExternalRuleLoader externalRuleLoader = RuboCopRulesDefinition.RULE_LOADER;
+    newExternalIssue
+      .type(externalRuleLoader.ruleType(ruleKey))
+      .severity(externalRuleLoader.ruleSeverity(ruleKey))
+      .remediationEffortMinutes(externalRuleLoader.ruleConstantDebtMinutes(ruleKey));
+
+    NewIssueLocation primaryLocation = newExternalIssue.newLocation()
+      .message(issue.message)
+      .on(inputFile);
+
+    if (issue.startLine !=null) {
+      boolean rangeIsProvided = issue.startColumn != null && issue.lastLine != null && issue.lastColumn != null;
+      boolean rangeIsValid = rangeIsProvided && (issue.startLine < issue.lastLine || (issue.startLine == issue.lastLine && issue.startColumn <= issue.lastColumn));
```

Comment:
The range could also be invalid if the file does not contain enough lines or if a line doesn't have enough columns. What do we do in such cases?

Reply:
`issues_when_rubocop_file_line_error` tests that we catch the exception: `100 is not a valid line for pointer. File useless-assignment.rb has 5 line(s)`. In this case we fail to import all the RuboCop report, but we continue the analysis. And it should never append.